demisto_sdk/commands/validate/validators/PB_validators/PB105_playbook_delete_context_all.py

```diff
@@ -0,0 +1,57 @@
+
+from __future__ import annotations
+
+from typing import Iterable, List
+
+from demisto_sdk.commands.content_graph.objects.playbook import Playbook
+from demisto_sdk.commands.validate.validators.base_validator import (
+        BaseValidator,
+        ValidationResult,
+)
+
+ContentTypes = Playbook
+
+
+class PlaybookDeleteContextAllValidator(BaseValidator[ContentTypes]):
+    error_code = "PB105"
+    description = ("Validate whether the playbook has a DeleteContext with all set to 'Yes'."
+                   " If the Playbook has it, it is not valid.")
+    rationale = "Playbook can not have DeleteContext script with arg all set to yes."
+    error_message = ("The playbook contains DeleteContext with all set to True which is not allowed."
+                     " For more info, see: https://xsoar.pan.dev/docs/playbooks/playbooks-overview#inputs-and-outputs")
+    related_field = "task"
+    is_auto_fixable = False
+
+    def if_delete_context_exists(self, playbook: ContentTypes):
+        """Check whether the playbook has a DeleteContext with all set to 'Yes'.
+            Args:
+                - content_items (Iterable[ContentTypes]): The content items to check.
+            Return:
+                - True if playbook has a DeleteContext with all set to 'Yes' and False otherwise.
+        """
+        tasks = playbook.tasks
+        for task in tasks.values():
+            curr_task = task.get("task", {})
+            script_args = task.get("scriptarguments", {})
+            if (
+                curr_task and curr_task.get("scriptName", "") == "DeleteContext"
+                and script_args and script_args.get("all", {}).get("simple", "") == "yes"
+            ):
+                return True
+        return False
```

Comment:
Let's return either an empty list or a list of all the task ids that uses DeleteContext=yes and make sure to list this ids in the error message.